core/src/main/java/org/ehcache/core/EhcacheBase.java

```diff
@@ -297,9 +297,6 @@ public V putIfAbsent(final K key, final V value) {
         if (put[0]) {
           putIfAbsentObserver.end(PutIfAbsentOutcome.PUT);
           return null;
-        } else if (inCache == null) {
```

Comment:
Other than the fact that this part of code was unnecessary, I was more interested in `Consumer<Boolean> put` to go, which is unlikely and we need it. Thus I still need to make `Store` spi change respectively. So lets wait for the Store layer refactoring.

Reply:
Oh! Wait a second. We actually might now that this if was removed. Silly me. Let me look.

Reply:
Done. No consumer anymore

Reply:
this helps